packages/cadl-ranch-specs/http/encode/datetime/main.tsp

```diff
@@ -0,0 +1,124 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+@doc("Test for encode decorator on datetime.")
+@supportedBy("dpg")
+@scenarioService("/encode/datetime")
+namespace Encode.Datetime;
+
+@encode(DateTimeKnownEncoding.rfc3339)
+scalar rfc3339Datetime extends utcDateTime;
+
+@route("/query")
+@scenario
+@scenarioDoc("""
+Test different encode for datetime query parameters.
+Expected query parameter: 
+default=2022-08-26T18:38:00.000Z
+rfc3339=2022-08-26T18:38:00.000Z
+rfc7231=Fri, 26 Aug 2022 14:38:00 GMT
+unix-timestamp=1686566864
+rfc3339-array=2022-08-26T18:38:00.000Z,2022-09-26T18:38:00.000Z
+""")
+op query(
+  @query
+  default: utcDateTime,
+
+  @query
+  @encode(DateTimeKnownEncoding.rfc3339)
+  rfc3339: utcDateTime,
+
+  @query
+  @encode(DateTimeKnownEncoding.rfc7231)
+  rfc7231: utcDateTime,
+
+  @query("unix-timestamp")
+  @encode(DateTimeKnownEncoding.unixTimestamp, int32)
```

Comment:
nit, I assume timestamp is usually long (aka int64)
https://learn.microsoft.com/en-us/dotnet/api/system.datetimeoffset.tounixtimeseconds?view=net-7.0